core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/arrayfunction/ArrayPrependExpression.kt

```diff
@@ -0,0 +1,36 @@
+package ch.ergon.dope.resolvable.expression.unaliased.type.arrayfunction
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.operator.FunctionOperator
+import ch.ergon.dope.validtype.ArrayType
+import ch.ergon.dope.validtype.ValidType
+
+class ArrayPrependExpression<T : ValidType>(
+    private val array: TypeExpression<ArrayType<T>>,
+    private val value: TypeExpression<T>,
+    private vararg val additionalValues: TypeExpression<T>,
+) : TypeExpression<ArrayType<T>>, FunctionOperator {
```

Comment:
Why cant you inherit from the _ArrayFunctionExpression_ here?

Reply:
As written in the description, PREPEND behaves syntactically different from APPEND, so this approach made the most sense to me in this case (and after syncing with @m0ne about it).